src/main/database.cpp

```diff
@@ -160,8 +162,8 @@ void DatabaseInstance::CreateMainDatabase() {
 	{
 		Connection con(*this);
 		con.BeginTransaction();
-		initial_database =
-		    db_manager->AttachDatabase(*con.context, info, config.options.database_type, config.options.access_mode);
+		AttachOptions options(config.options.access_mode, config.options.database_type);
```

Comment:
Maybe the block size should also be in the config so that we can set it for the main database and not only for `ATTACHed` databases